### ISSUE TITLE
Implement Additive Attention

### ISSUE THREAD

**Issue by Rishit-dagli** [Rishit-dagli/Fast-Transformer#4]
Implement Additive Attention as a TensorFlow layer:
- [x] Figure out using rotary embeddings
- [x] Add masking functionality
- [x] Relative Position embeddings
- [x] Calculate query attention logits
- [x] Calculate Global Query tokens
- [x] Calculate key attention logits
- [x] Calculate Global Key tokens
- [x] Add queries as residuals
